### PR TITLE
Style type decision2

### PR DESCRIPTION
**Pull request by index-git**
- [x] Tests
- [ ] ~~Layman Test Client~~
- [x] Changlelog
- [ ] ~~Documentation~~

### CODE REVIEW

File: src/layman/layer/filesystem/input_style.py [hunk 1]
```diff
@@ -85,18 +83,16 @@ def get_metadata_comparison(username, publication_name):
...
+    if file_storage:
+        xml = file_storage.read()
+        file_storage.seek(0)
+        xml_tree = etree.fromstring(xml)
+        root_tag = xml_tree.tag
         root_attribute = etree.QName(root_tag).localname
         result = next((sd for sd in layer.STYLE_TYPES_DEF if sd.root_element == root_attribute), None)
         if not result:
             raise LaymanError(46)
     else:
```
Comment (jirik) [inline src/layman/layer/filesystem/input_style.py:96]: Probably redundant

File: src/layman/layer/rest_layer.py [hunk 2]
```diff
@@ -110,16 +117,22 @@ def patch(username, layername):
...
             try:
                 style_file = deleted['style']['file']
+                style_type = input_style.get_style_type_from_file_storage(style_file)
+                kwargs['style_type'] = style_type
+                kwargs['store_in_geoserver'] = style_type.store_in_geoserver
             except KeyError:
                 pass
         if style_file is not None:
-            input_style.save_layer_file(username, layername, style_file)
+            style_type = input_style.get_style_type_from_file_storage(style_file)
```
Comment (jirik) [inline src/layman/layer/rest_layer.py:126]: Lines are not needed:
```
style_type = input_style.get_style_type_from_file_storage(style_file)
            kwargs['style_type'] = style_type
            kwargs['store_in_geoserver'] = style_type.store_in_geoserver
```

File: src/layman/layer/rest_layers_test.py [hunk 3]
```diff
@@ -47,31 +48,41 @@ def assert_style_file(workspace,
...
+                               expected_style_file_template,
+                               expected_style_type):
     workspace = 'test_style_correctly_saved_workspace'
     layer = 'test_style_correctly_saved_layer' + layer_suffix
     expected_style_file = expected_style_file_template.format(workspace=workspace, layer=layer) if expected_style_file_template else None
     process_client.publish_layer(workspace,
                                  layer,
                                  style_file=source_style_file_path)
     assert_style_file(workspace, layer, expected_style_file)
+    info = process_client.get_layer(workspace, layer)
```
Comment (jirik) [inline src/layman/layer/rest_layers_test.py:69]: Do not call process_client.get_layer, call util.get_publication_info and remove style_type from response

File: src/layman/layer/rest_layers_test.py [hunk 4]
```diff
@@ -47,31 +48,41 @@ def assert_style_file(workspace,
...
+    assert info['style_type'] == expected_style_type
+
+    process_client.delete_layer(workspace, layer)
+    process_client.publish_layer(workspace, layer)
 
     with app.app_context():
-        input_style.delete_layer(workspace, layer)
+        info = process_client.get_layer(workspace, layer)
```
Comment (jirik) [inline src/layman/layer/rest_layers_test.py:76]: Do not call process_client.get_layer, call util.get_publication_info

File: src/layman/layer/rest_layers_test.py [hunk 5]
```diff
@@ -47,31 +48,41 @@ def assert_style_file(workspace,
...
+        assert info['style_type'] == 'sld'
+
     assert_style_file(workspace, layer, None)
 
     process_client.patch_layer(workspace,
                                layer,
                                style_file=source_style_file_path)
     assert_style_file(workspace, layer, expected_style_file)
+    info = process_client.get_layer(workspace, layer)
```
Comment (jirik) [inline src/layman/layer/rest_layers_test.py:85]: Do not call process_client.get_layer, call util.get_publication_info

**Review by jirik — CHANGES_REQUESTED** [review]

File: src/layman/layer/filesystem/input_style.py (hunk 1 above)
Comment (index-git) [reply src/layman/layer/filesystem/input_style.py:96]: It is not and I like, that the logic 'no style file means SLD style' is at one place.

**Review by jirik — APPROVED** [review]

**Pull request merged**